transmission/transmission.go

```diff
@@ -387,8 +387,11 @@ func (b *batchAgg) fireBatch(events []*Event) {
 
 		var req *http.Request
 		reqBody, zipped := buildReqReader(encEvs, !b.disableCompression)
-		req, err = http.NewRequest("POST", url.String(), reqBody)
-		req.ContentLength = int64(reqBody.Len())
+		if reader, ok := reqBody.(*pooledReader); ok {
+			req, err = http.NewRequest("POST", url.String(), &reader.Reader)
+		} else {
+			req, err = http.NewRequest("POST", url.String(), reqBody)
+		}
```

Comment:
Why is this type switch necessary? You shouldn't need to pass the embedded Reader directly.

Reply:
because of https://go.dev/play/p/mNGgU5JkY1G sob